Kuznetsov Artyom (Laboratory work №1 option №1)

Kuznetsov-Artyom opened the pull request:
A plugin has been implemented that outputs information about user-defined data types, as well as information about members.
Example:
```cpp
struct Point3D {
  double m_x{};
  double m_y{};
  double m_z{};
};

template <typename T> struct NodeList {
  T data{};
  NodeList<T> *next{};
};

template <typename T> class CheckStaticConstTemplate {
  static const T m_member;
};

template <typename T> const T CheckStaticConstTemplate<T>::m_member{};
```
Result:
```
Point3D (struct)
|_ m_x (double|public)
|_ m_y (double|public)
|_ m_z (double|public)

NodeList (struct|template)
|_ data (T|public)
|_ next (NodeList<T> *|public)

CheckStaticConstTemplate (class|template)
|_ m_member (const T|private|static)
```

Kuznetsov-Artyom committed:
create folders

Kuznetsov-Artyom committed:
add visitor

Kuznetsov-Artyom committed:
add cosumer

Kuznetsov-Artyom committed:
add plugin action & plugin registry

Kuznetsov-Artyom committed:
upgrade task

Kuznetsov-Artyom committed:
done

Kuznetsov-Artyom committed:
Merge branch 'course-spring-2024' into dev

m-ly4 commented on clang/test/lab1/kuznetsov_artyom/test.cpp:
```diff
@@ -0,0 +1,62 @@
+    // CHECK-NEXT: |_ m_cash (int|public)
+    int m_cash{};
+  };
+};
```
Could you please add also member field of type `Human` into `User` definition?

m-ly4 commented on clang/labs/lab1/kuznetsov_artyom/PrintClassMembers.cpp:
```diff
@@ -0,0 +1,87 @@
+    if (declaration->isStruct() || declaration->isClass()) {
+      outInfoUserType(declaration);
+
+      auto members = declaration->fields();
```
Could you please also print `static` attribute of members?

Kuznetsov-Artyom committed:
add static member test & const member test

Kuznetsov-Artyom committed:
update

Kuznetsov-Artyom commented:
I have updated the description for the commit.

Added:
1) Output of static class members
2) The Human member in the User class
3) Other tests

Other:
1) Special output for a static class member (see the description)

m-ly4 reviewed (approved):
Thank you for applying suggestions.
